x/bep3/simulation/decoder.go

```diff
@@ -1,12 +1,37 @@
 package simulation
 
 import (
+	"bytes"
+	"fmt"
+
 	"github.com/cosmos/cosmos-sdk/codec"
 	cmn "github.com/tendermint/tendermint/libs/common"
+
+	"github.com/kava-labs/kava/x/bep3/types"
 )
 
 // DecodeStore unmarshals the KVPair's Value to the module's corresponding type
 func DecodeStore(cdc *codec.Codec, kvA, kvB cmn.KVPair) string {
-	// TODO implement this
-	return ""
+	switch {
+	case bytes.Equal(kvA.Key[:1], types.AtomicSwapKeyPrefix):
+		var swapA, swapB *types.AtomicSwap
```

Comment:
Is this meant to be just `types.AtomicSwap`, not a pointer?

Reply:
EDIT: you were right. Idk why I thought that this was a pointer. Sorry about that